### ISSUE TITLE
MinGW GCC on Window

### ISSUE THREAD

**Issue by JG-Adams** [Madsycode/book-fuse-engine#5]
The awesome tutorial! :)
I'm using VSCode and MSYS2 MinGW.
This use MSVC for window. but, I prefer using GCC. So, I wanna get it to work.

First step in the right direction is:
Change this order of linking: SDL2 SDL2main
Into: mingw32 SDL2main SDL2
Then replace the bin and libs binary files with MinGW64 version.

It was looking for yaml_cppd but the generated from source gave me libyaml_cpp.dll
I made it look for yaml_cpp instead. Now it compile successfully.

I run the program. Nothing happened. Is this normal?

Edit:
I got it to run the game!
I looked at the bin folder, it had the game.exe and engine. Then I noticed the rest of the binaries dlls were in Debug folder.
Separate from the Game.exe. I need explanation for how this is supposed to work. Still reading the book.
Thanks! :)

**Comment by Madsycode** [comment]
Am not sure because the only time I used MinGW was with Codeblock so I can't really help there! but if nothing is happening, then it's becuse it can't find the DLL. The way you can be sure about it is by trying to execute the .exe file by double clicking on it, which will triggere a runtime assertion telling you which .dll file is missing!

**Comment by JG-Adams** [comment]
Thank you, that's what fixed it.
But, when I build the program it place the game in bin folder and all Binaries and Resources inside the bin/Debug.
Then I do F5 and it tries to run the game from the Debug folder. But, it don't exist unless I move it there.
Is it supposed to run from Bin? I want to know how it was supposed to work.

I could fix it by doing this 
set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin/Debug)
set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin/Debug)
set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin/Debug)
instead of bin.

But, is it supposed to run from bin? If it does, I want to know the purpose of it. And try to make it work.
I don't usually work with Debug mode. So, this is new to me.
Any help is appreciated!

**Comment by Madsycode** [comment]
You can simply modify the launch.json 'program' attribute to your .exe file path! and regarding the set(CMAKE_...) you can force cmake to always use the path as specified regardless of the platform like this:  
set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "$<0:>${CMAKE_BINARY_DIR}/bin/Debug)" the $<0:> tells cmake to use the path as specified because in linux, the Debug folder is not generate by default as in Windows! keep figthing it's the only way to learn! don't give you will get there!

**Comment by JG-Adams** [comment]
Thanks! I already solved it. Not sure I understand the $<0:> use.
I just have question about the purpose of a Debug folder that has everything when there is a bin that don't.
The Master code is left this way.
Basically, I don't see the use of a folder Debug when I can put everything in Bin.
How it's meant to be used I'd like to know. :)

I gotta say, I've been learning C++ by myself for about 5 years on and off and really struggled with the build more than anything.
When I tried your book. I learned of a much better way to do it now. Thank you very much Sir! :)
This helped me to get into more coding and less configuring! :)
